featurebyte/feast/service/registry.py

```diff
@@ -223,7 +241,9 @@ async def _populate_registry(self, document: FeastRegistryModel) -> FeastRegistr
 
     async def _move_registry_to_storage(self, document: FeastRegistryModel) -> FeastRegistryModel:
         feast_registry_path = Path(f"feast_registry/{document.id}/feast_registry.pb")
-        await self.storage.put_bytes(document.registry, feast_registry_path)
+        with self.get_registry_storage_lock(timeout=FEAST_REGISTRY_REDIS_LOCK_TIMEOUT):
```

Comment:
I think we may need to get the lock earlier before getting the enabled feature lists.
Both `_get_online_enabled_feature_lists` and  `_create_or_update_feast_registry` needs to be done after acquiring the lock and completed before releasing

Reply:
I don't fully get the idea behind this. The main purpose here is to prevent concurrent update of the registry path. If we want to extend it to the feature list update, how about feature update?

Reply:
Or we delegate the feature update to the offline table Redis lock? If that is the case, i think it makes sense.